src/node_buffer.cc

```diff
@@ -949,6 +983,8 @@ void IndexOfBuffer(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_UNLESS_BUFFER(Environment::GetCurrent(args), args[0]);
```

Comment:
sure, i can add that, but how did this work before? `IndexOfBuffer` was used by `indexOf` before this PR and didn't do the check